src/app.rs

```diff
@@ -149,6 +152,23 @@ impl Application {
                 let mut config =
                     config::load_from_paths(&config_paths, false).map_err(handle_config_errors)?;
 
+                // Conditionally wire up the 'tap' controller, based on API availability
+                let tap_controller = if cfg!(feature = "api") {
+                    let tc = TapController::new();
+
+                    let tap_sinks = config
+                        .sources
+                        .iter()
+                        .map(|(name, _)| (format!("_tap_{}", &name), tc.make_sink(name)))
```

Comment:
This manually connects a `tap` sink to each of the source/transform inputs. Note that the format used is currently:

```
_tap_{component_name}
```

This can result in clashes in the (unlikely) event that a user has named their component `_tap_{existing_other_name}`.

One option may be to disallow `_` as a leading character of a component name, and designate anything with it as an internal sink. This would be a breaking change, though.

cc @lukesteensen @jszwedko for thoughts.